okhttp-clients/src/test/java/com/palantir/conjure/java/okhttp/CatchThrowableInterceptorTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.okhttp;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import okhttp3.Interceptor;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class CatchThrowableInterceptorTest {
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS) private Interceptor.Chain chain;
+
+    @Test
+    public void rethrowsIoException() throws IOException {
+        IOException exception = new IOException();
+        when(chain.proceed(chain.request())).thenThrow(exception);
+        assertThatExceptionOfType(IOException.class)
+                .isThrownBy(() -> CatchThrowableInterceptor.INSTANCE.intercept(chain))
+                .isEqualTo(exception);
+    }
+
+    @Test
+    public void stackOverflowErrorIsWrapped() throws IOException {
+        StackOverflowError error = new StackOverflowError();
+        when(chain.proceed(chain.request())).thenThrow(error);
+        assertThatExceptionOfType(IOException.class)
+                .isThrownBy(() -> CatchThrowableInterceptor.INSTANCE.intercept(chain))
+                .withCause(error);
+    }
```

Comment:
May want to add something like this to verify that it does actually fail gracefully in practice:
https://github.com/palantir/conjure-java-runtime/pull/926/files

Reply:
Eh that test is outdated, tracer has been refactored to catch+log since then. disregard.